### ISSUE TITLE
[studio] Publish all fails when blob is missing with no feedback

### ISSUE THREAD

**Issue by avasquez614** [craftercms/craftercms#5949]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

When you do a publish all and a blob is missing from S3 (but the blob reference is in the repository and being published), the entire publish fails, and no indication of the error is specified in the UI or the logs.

This is specially a problem for clients that have huge repositories, and when synching down from an upper environment, a .blob store reference is pulled down but the S3 file is missing.

### Steps to reproduce

Steps:

1. Configure blob store in a site
2. Upload a blo through Studio so the blob is stored in S3 and the .blob is created in the sandbox repository
3. Delete the file directly from S3
4. Attempt a publish all.

From the UI side, a red popup appears but nothing else is shown. The publisher says it's ready instead of being disabled and indicating that there was an issue.

From the logs an AWS error is printed out but it doesn't indicate the actual file with the problem.

It would be nice if the publish all successfully continued and just skipped the file with the error and indicated that in the UI and the logs. If not, at least indicating the error in the UI and logs should suffice

### Relevant log output

```shell
https://gist.github.com/avasquez614/b09e54069f93fb79645b9afbcb475074
```

**Comment by yacdaniel** [comment]
I tried the steps and `Publish Everything` works with the error, but the next publish methods are not responding correctly.

- Publish by commit, the logs are
```
[INFO] 2023-03-03T10:19:52,467 [studio.clockTaskExecutor-18] [job.StudioPublisherTask] | Publish started in site 'empty12345' for target 'live' with '1' items ready to be published 
[INFO] 2023-03-03T10:19:52,469 [studio.clockTaskExecutor-18] [job.StudioPublisherTask] | Publish '1' items in site 'empty12345' to target 'live' 
[WARN] 2023-03-03T10:19:52,775 [studio.clockTaskExecutor-18] [deployment.PublishingManagerImpl] | Item in site 'empty12345' path '/static-assets/s3/banquero.png' doesn't exist in the database nor the git repository. Skipping publishing of this item. 
[INFO] 2023-03-03T10:19:52,850 [studio.clockTaskExecutor-18] [job.StudioPublisherTask] | Published '1' items in site 'empty12345' to target 'live' 
```

1- Item exist in repo and db but is missing from blob store, should log message be like the one from a failure in `Publish All` ?  
2- The `Publish Status` remains as  `Ready` should this be `ReadyWithErrors` too? 

- Publish by path, there is an error here
https://gist.github.com/yacdaniel/41f198a000a19377af4c4b383246e5d7

3 - `Publish Status`  goes to `disabled` because of the error, all items that can be published should continue and fail/log the error for the missing blob item?, same as with `Publish Everything` ? 

What should be the correct behaivor for those publish mehtods/errors @sumerjabri ?

**Comment by jmendeza** [comment]
Updated publisher behavior to update item states only after successful publishing of each package.
So now the publisher will stop and items in the failing list would continue to appear as modified/new.

**Comment by yacdaniel** [comment]
Verified fixed in 4.0.3-SNAPSHOT.
